src/assets/wise5/services/translateProjectService.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { ProjectService } from './projectService';
+import { ConfigService } from './configService';
+import { Observable, of, tap } from 'rxjs';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { copy } from '../common/object/object';
+
+@Injectable()
+export class TranslateProjectService {
+  constructor(
+    private configService: ConfigService,
+    private http: HttpClient,
+    private projectService: ProjectService
+  ) {}
+
+  translate(locale = 'en_US'): Observable<any> {
+    const project = this.revertToOriginalProject();
+    return this.projectService.getLocale().hasTranslationsToApply(locale)
+      ? this.http
+          .get(this.getTranslationMappingURL(locale), {
+            headers: new HttpHeaders().set('cache-control', 'no-cache')
+          })
+          .pipe(
+            tap((translations: any) => {
+              this.applyTranslations(project, translations);
+              this.projectService.setProject(project);
+            })
+          )
+      : of({});
```

Comment:
Yes, we can, but an issue may arise when we add code in the future that calls translate() from InitializeVLEService.initialize[Student|Preview]() while the Student/Preview views are bootstrapping. In this case, I was imagining that we'd want to do these in order:
1. Retrieve project (subscribe to this call to make sure project is retrieved before moving on)
2. Translate project (subscribe to this call to make sure project is translated before moving on)
3. Initialize everything else

To make the above possible, this translate() function would need to return an Observable (so we can subscribe to it in InitializeVLEService.initialize[Student|Preview]()). 

The other option is to say that the translation in step 2 does not have to happen before step 3, that it can run as part of the "Initialize everything else" sequence. I'm not sure what the ramifications of that would be on the application (e.g. will it cause the page to flicker?), but we can try it?

Reply:
As we discussed as a group, I changed the translate() func to return a Promise instead of an Observable. This removed the need to ```subscribe()```. This should also let us ```await``` for it in InitializeVLEService in the future.